pymoskito/simulation_modules.py

```diff
@@ -72,60 +72,23 @@ def __init__(self, settings):
         self._settings["step width"] = None
         self._settings.pop("modules", None)
 
-    @property
-    @abstractmethod
-    def public_settings(self):
-        pass
-
-    @property
-    def settings(self):
-        return self._settings
-
-    @property
-    def tick_divider(self):
-        return self._settings["tick divider"]
-
-    @property
-    def step_width(self):
-        return self._settings["step width"]
+        if settings.get("binding", False):
+            self._generateBinding(settings)
 
-    @step_width.setter
-    def step_width(self, value):
-        self._settings["step width"] = value
-
-    @abstractmethod
-    def calc_output(self, input_vector):
-        pass
-
-
-class CppSimulationModule(QObject):
-    """
-    Base class for a cpp simulation module.
-
-    """
-
-    def __init__(self, settings):
-        QObject.__init__(self, None)
+    def _generateBinding(self, settings):
```

Comment:
This method only depends on `settings["Module"]` and therefore may be refactored as a function in the `tools` module.